graphql_compiler/post_processing/sql_post_processing.py

```diff
@@ -24,20 +24,32 @@ def _mssql_xml_path_string_to_list(
     if xml_path_result == "":
         return []
 
+    # Some of the special characters involved in XML path array aggregation.
+    delimiter = "|"
+    null = "~"
+
+    # Remove the "|" from the first result in the string representation of the list.
+    if xml_path_result[0] != delimiter:
+        raise AssertionError(
+            f"Unexpected fold result. All XML path array aggregated lists must start with a '|'. "
```

Comment:
nit: `must start with a '{delimiter}'` instead of another hardcoded `|`,  in case it changes in the future